packages/web3-utils/test/fixtures/validation.ts

```diff
@@ -28,41 +28,43 @@ export const compareBlockNumbersValidData: [[Numbers, Numbers], number][] = [
 	[[1, BigInt(1)], 0],
 	[[1, BigInt(2)], -1],
 	[[2, BigInt(1)], 1],
-
 	[['earliest', 0], 0],
+	[[0, 'earliest'], 0],
+	[['earliest', 'earliest'], 0],
 	[['pending', 'pending'], 0],
 	[['latest', 'latest'], 0],
-	[['earliest', 2], -1],
 	[['earliest', 'pending'], -1],
 	[[BigInt('9007199254740992'), BigInt('9007199254740991')], 1],
 	[[13532346, 13532300], 1],
 	[['pending', 'latest'], 1],
-	[['latest', 0], 1],
-	[['latest', BigInt(1)], 1],
-	[['pending', 0], 1],
-	[['pending', BigInt(1)], 1],
 	[['safe', 'safe'], 0],
 	[['earliest', 'safe'], -1],
-	[['safe', 0], 1],
-	[[0, 'safe'], -1],
 	[['safe', 'pending'], -1],
 	[['pending', 'safe'], 1],
 	[['finalized', 'finalized'], 0],
 	[['earliest', 'finalized'], -1],
-	[['finalized', 0], 1],
 	[['finalized', 'pending'], -1],
```

Comment:
Bring back some deleted tests as they will be valid again if the suggestion to code was applied.
```suggestion
    [['earliest', 'finalized'], -1],
	[['finalized', 0], 1],
	[['finalized', 'pending'], -1],
	[[0, 'finalized'], -1],
```

Reply:
shared reason above, in some slim edge case finalized can be 0.